### ISSUE TITLE
Update to IDF 5.0

### ISSUE THREAD

**Issue by persello** [pulse-loop/bluedroid#12]

**Comment by dberlin** [comment]
I made the changes to do this here: #14 
The main annoying issue is that esp-idf-svc requires a patch that is only available in git
(They have a strong desires to release it in lockstep with some other esp-idf libraries)

**Comment by persello** [comment]
Thanks for the contribution! I plan to do a new release of bluedroid that will include your PR, as soon that version of esp-idf-svc is released on crates.io (maybe I'll release v0.4, so we can keep patching the 0.3 branch for those who are on ESP-IDF 4).

**Comment by dberlin** [comment]
I have a version that works with the upcoming 5.1 release which adds a controller field and changes a constant.  I also added cfg blocks (esp-idf provides them) so it works on all versions (4,5,5.1) without breaking anything.  Since this might be little till you can release I'll update this PR.

**Comment by persello** [comment]
That's really useful, I didn't know about `#[cfg]` attributes applied to ESP-IDF versions. I'll have a look in the following days. If you can update this PR, it would be really useful.

**Comment by dberlin** [comment]
Absolutely, i'm on it.

**Comment by dberlin** [comment]
The updated pull request should now work on 5.0, 5.1, and 4.x

4.x is too old to work by default on my fedora 38 laptop (python in fedora 38 is too new), i'll set up a virtual environment to test it in

I have tested it on 5.0 and 5.1 (current release branches of each)

**Comment by persello** [comment]
This crate is getting merged into `esp-idf-svc`, so I’ll be working on setting up a few automatic checks with GitHub Actions to test that the crate continuously compiles for multiple IDF versions, before starting to work on their pull request.
